yt_dlp/cookies.py

```diff
@@ -192,68 +195,158 @@ def _extract_firefox_cookies(profile, container, logger):
+        }[browser_name]
 
     else:
-        yield from map(os.path.expanduser, (
-            '~/.mozilla/firefox',
-            '~/snap/firefox/common/.mozilla/firefox',
-            '~/.var/app/org.mozilla.firefox/.mozilla/firefox',
-        ))
+        flatpak_root = os.path.expanduser('~/.var/app')
+        snap_root = os.path.expanduser('~/snap')
+        browser_dirs = {
+            'firefox': [
+                os.path.expanduser('~/.mozilla/firefox'),
+                os.path.join(flatpak_root, 'org.mozilla.firefox/.mozilla/firefox'),
+                os.path.join(snap_root, 'firefox/common/.mozilla/firefox'),
+            ],
+            'librewolf': [
+                os.path.expanduser('~/.librewolf'),
+                os.path.join(flatpak_root, 'io.gitlab.librewolf-community/.librewolf'),
+                # not published on snapcraft
+            ],
+        }[browser_name]
 
+    return _FirefoxBrowserSettings(
+        browser_dirs=browser_dirs,
+    )
 
 def _firefox_cookie_dbs(roots):
     for root in map(os.path.abspath, roots):
         for pattern in ('', '*/', 'Profiles/*/'):
             yield from glob.iglob(os.path.join(root, pattern, 'cookies.sqlite'))
 
 
+@dataclass
+class _ChromiumBrowserSettings:
+    browser_dirs: list[str]
+    keyring_name: str
+    keyring_application_name: str
+    supports_profiles: bool
+
+    @property
+    def mac_keyring_account(self) -> str:
+        return self.keyring_name
+
+    @property
+    def mac_keyring_service(self) -> str:
+        return f'{self.keyring_name} Safe Storage'
+
+    @property
+    def kwallet_password(self) -> str:
+        return f'{self.keyring_name} Safe Storage'
+
+    @property
+    def kwallet_folder(self) -> str:
+        return f'{self.keyring_name} Keys'
+
+    @property
+    def gnome_keyring_application_name(self) -> str:
+        return self.keyring_application_name
+
+    @property
+    def gnome_keyring_label(self) -> str:
+        return f'{self.keyring_name} Safe Storage'
+
+
 def _get_chromium_based_browser_settings(browser_name):
     # https://chromium.googlesource.com/chromium/src/+/HEAD/docs/user_data_dir.md
     if sys.platform in ('cygwin', 'win32'):
         appdata_local = os.path.expandvars('%LOCALAPPDATA%')
         appdata_roaming = os.path.expandvars('%APPDATA%')
-        browser_dir = {
-            'brave': os.path.join(appdata_local, R'BraveSoftware\Brave-Browser\User Data'),
-            'chrome': os.path.join(appdata_local, R'Google\Chrome\User Data'),
-            'chromium': os.path.join(appdata_local, R'Chromium\User Data'),
-            'edge': os.path.join(appdata_local, R'Microsoft\Edge\User Data'),
-            'opera': os.path.join(appdata_roaming, R'Opera Software\Opera Stable'),
-            'vivaldi': os.path.join(appdata_local, R'Vivaldi\User Data'),
-            'whale': os.path.join(appdata_local, R'Naver\Naver Whale\User Data'),
+        browser_dirs = {
```

Comment:
I tried installing a few browsers (brave and opera and edge) through the microsoft store and unlike firefox I think these installations are just like regular installations so there is no separate Packages path to search in